libs/gretty/src/main/groovy/org/akhikhl/gretty/ServerConfigWithInputs.java

```diff
@@ -0,0 +1,172 @@
+package org.akhikhl.gretty;
+
+import groovy.lang.Closure;
+import org.gradle.api.model.ReplacedBy;
+import org.gradle.api.tasks.Console;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
+
+import java.util.List;
+import java.util.Map;
+
+public interface ServerConfigWithInputs {
+    @Optional
+    @Input
+    List<String> getJvmArgs();
+
+    @Optional
+    @Input
+    Map<String, String> getSystemProperties();
+
+    @Optional
+    @Input
+    String getServletContainer();
+
+    @Optional
+    @Input
+    Boolean getManagedClassReload();
+
+    @Optional
+    @Input
+    String getHost();
+
+    @Optional
+    @Input
+    Boolean getHttpEnabled();
+
+    @Optional
+    @Input
+    Integer getHttpPort();
+
+    @Optional
+    @Input
+    Integer getHttpIdleTimeout();
+
+    @Optional
+    @Input
+    Boolean getHttpsEnabled();
+
+    @Optional
+    @Input
+    Integer getHttpsPort();
+
+    @Optional
+    @Input
+    Integer getHttpsIdleTimeout();
+
+    @Optional
+    @Input
+    String getSslHost();
+
+    @Optional
+    @InputFile
+    @PathSensitive(PathSensitivity.NONE)
+    Object getSslKeyStorePath();
+
+    @Internal
+    String getSslKeyStorePassword();
```

Comment:
Is this a security concern? If that password is not part of the inputs and then changed, the task might not re-execute, is that correct?

Reply:
Passwords should never be inputs. The output of the task does not depend on the password. Only whether the task fails or not depends on the password. And yes, this is also a security concern.

Reply:
That's interesting, never thought of it that way. When thinking more, this also cannot fail: likely changing the password also changes the file (due to encryption with the new password), which is again tracked by Gradle, right?

Reply:
Yes.